src/shared/sam/debugger/awsSamDebugConfigurationValidator.ts

```diff
@@ -0,0 +1,180 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { samLambdaRuntimes } from '../../../lambda/models/samLambdaRuntime'
+import { CloudFormation } from '../../cloudformation/cloudformation'
+import { CloudFormationTemplateRegistry } from '../../cloudformation/templateRegistry'
+import { localize } from '../../utilities/vsCodeUtils'
+import {
+    AWS_SAM_DEBUG_REQUEST_TYPES,
+    AWS_SAM_DEBUG_TARGET_TYPES,
+    AwsSamDebuggerConfiguration,
+    CODE_TARGET_TYPE,
+    isCodeTargetProperties,
+    isTemplateTargetProperties,
+    TemplateTargetProperties
+} from './awsSamDebugConfiguration'
+
+export interface ValidationResult {
+    isValid: boolean
+    message?: string
+}
+
+export interface AwsSamDebugConfigurationValidator {
+    validateSamDebugConfiguration(debugConfiguration: AwsSamDebuggerConfiguration): ValidationResult
+    isValidSamDebugConfiguration(debugConfiguration: AwsSamDebuggerConfiguration): boolean
+}
+
+export class DefaultAwsSamDebugConfigurationValidator implements AwsSamDebugConfigurationValidator {
```

Comment:
I really don't think this needs to be mocked. So the `AwsSamDebugConfigurationValidator` interface can be removed. It's just logic.